File: mobile-legacy/src/data/repositories/ApiAuthRepository.ts
import { AuthRepository } from '../../domain/repositories/AuthRepository';
import { User } from '../../domain/entities/User';
import { Platform } from 'react-native';
import { Config } from '../../config';

const API_BASE_URL = `${Config.JAVA_API_URL}/api/v1/users`;

export class ApiAuthRepository implements AuthRepository {
  
  async login(email: string, password: string): Promise<User> {
    try {
      const response = await fetch(`${API_BASE_URL}/login`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ email, password }),
      });

      if (!response.ok) {
        const errorData = await response.json().catch(() => null);
        throw new Error(errorData?.message || 'Login failed.');
      }

      const data = await response.json();
      
      // Java API, User objesini döndürüyor. Biz de kendi mobil User arayüzümüze mapliyoruz.
      return {
        id: data.id.toString(),
        email: data.email,
        fullName: data.fullName,
        profilePictureUrl: data.profilePictureUrl,
      };
    } catch (error) {
      console.error('Login Error:', error);
      throw error;
    }
  }

  async loginWithGoogle(idToken: string): Promise<User> {
    return this.loginWithSocial(idToken, 'GOOGLE');
  }

  async loginWithSocial(idToken: string, platform: string): Promise<User> {
    try {
      const response = await fetch(`${API_BASE_URL}/login/social`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ idToken, platform }),
      });

      if (!response.ok) {
        const errorData = await response.json().catch(() => null);
        throw new Error(errorData?.message || `${platform} giriş işlemi başarısız oldu.`);
      }

      const data = await response.json();
      return {
        id: data.id.toString(),
        email: data.email,
        fullName: data.fullName,
        profilePictureUrl: data.profilePictureUrl,
      };
    } catch (error) {
      console.error(`${platform} Login Error:`, error);
      throw error;
    }
  }

  async logout(): Promise<void> {
    // JWT/Token temizleme işlemi gerektiğinde buraya eklenebilir.
    return Promise.resolve();
  }

  getCurrentUser(): User | null {
    // Mobil tarafta state yönetimi Zustand üzerinden yapıldığı için, repository kendi State'ini tutmuyor.
    return null;
  }

  async deleteAccount(): Promise<void> {
    try {
      const response = await fetch(`${API_BASE_URL}/me`, {
        method: 'DELETE',
        headers: {
          'Content-Type': 'application/json',
          // Note: Auth token is usually handled by a wrapper or interceptor, 
          // but here we assume the fetch call in context of an authenticated session.
        },
      });

      if (!response.ok) {
        const errorData = await response.json().catch(() => null);
        throw new Error(errorData?.message || 'Hesap silme işlemi başarısız oldu.');
      }
    } catch (error) {
      console.error('Delete Account Error:', error);
      throw error;
    }
  }
}

File: mobile-android/app/src/main/java/com/antigravity/mobile/data/repository/ApiAuthRepository.kt
package com.antigravity.mobile.data.repository

import com.antigravity.mobile.domain.model.User
import com.antigravity.mobile.domain.repository.AuthRepository
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.http.ContentType
import io.ktor.http.contentType
import kotlinx.serialization.Serializable
import javax.inject.Inject

@Serializable
data class SocialLoginRequest(
    val idToken: String,
    val platform: String
)

class ApiAuthRepository @Inject constructor(
    private val client: HttpClient
) : AuthRepository {

    private val baseUrl = "http://10.0.2.2:8080/api/v1/users" // Emulator localhost for Java Core API

    override suspend fun loginWithGoogle(idToken: String): Result<User> {
        return try {
            val response = client.post("$baseUrl/login/social") {
                contentType(ContentType.Application.Json)
                setBody(SocialLoginRequest(idToken, "GOOGLE"))
            }.body<User>()
            Result.success(response)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    override suspend fun logout(): Result<Unit> {
        // Implementation for clearing local tokens can be added here
        return Result.success(Unit)
    }

    override suspend fun getCurrentUser(): User? {
        // Future implementation with local storage
        return null
    }

    override suspend fun deleteAccount(): Result<Unit> {
        return try {
            client.post("$baseUrl/me/delete")
            Result.success(Unit)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}
